### ISSUE TITLE
expect raw []byte to be encoded as is

### ISSUE THREAD

**Issue by Gaudeamus** [bet365/jingo#6]
Hello, thanx for the quick fix for nil struct member.
Got another issue

```
type testStruct2 struct {
	Raw		[]byte 	`json:"raw"`
}

func TestJsoniterEnc2(t *testing.T) {
	var jingoEnc2 = jingo.NewStructEncoder(testStruct2{})
	buf := jingo.NewBufferFromPool()
	v := testStruct2{
		Raw:[]byte(`{"mapKey1":1,"mapKey2":2}`),
	}

	jingoEnc2.Marshal(&v, buf)
	fmt.Print(buf)
	buf.ReturnToPool()
}
```
out:
```
{"raw":[123,34,109,97,112,75,101,121,49,34,58,49,44,34,109,97,112,75,101,121,50,34,58,50,125]}
```
while obvious to get:
```
{"raw":{"mapKey1":1,"mapKey2":2}}
```

**Comment by Gaudeamus** [comment]
also tried `stringer`
```
type rawBytes []byte

func (b rawBytes) String() string {
	if b == nil {
		return "null"
	}
	if len(b)== 0 {
		return "[]"
	}
	s := strconv.Quote(string(b))
	return s[1:len(s)-1]
	return fmt.Sprintf("%q", string(b))
}
type testStruct2 struct {
	Raw		rawBytes 	`json:"raw,stringer"`
	Raw1	rawBytes 	`json:"raw1,stringer"`
	Raw2    []byte 	`json:"raw2"`
}

func TestJingoEnc2(t *testing.T) {
	var jingoEnc2 = jingo.NewStructEncoder(testStruct2{})
	buf := jingo.NewBufferFromPool()
	buf.Reset()
	vd := testStruct2{}
	v := testStruct2{
		Raw:[]byte(`{"mapKey1":1,"mapKey2":2}`),
		Raw2:[]byte(`{"mapKey1":1,"mapKey2":2}`),
	}
	jingoEnc2.Marshal(&v, buf)
	fmt.Println(buf)
	json.Unmarshal(buf.Bytes, &vd)
	fmt.Printf("%+v\n",vd)
	buf.ReturnToPool()
}
```

out:
```
{"raw":"{\"mapKey1\":1,\"mapKey2\":2}","raw1":"null","raw2":[123,34,109,97,112,75,101,121,49,34,58,49,44,34,109,97,112,75,101,121,50,34,58,50,125]}

{Raw:null Raw1:\x9e\xe9e Raw2:[123 34 109 97 112 75 101 121 49 34 58 49 44 34 109 97 112 75 101 121 50 34 58 50 125]}
```
so it seems impossible to proper handle []byte

**Comment by Gaudeamus** [comment]
also does not perform escaping:
```
func TestJingoEnc3(t *testing.T) {
	type testStruct3 struct {
		Str		string `json:"str"`
	}

	var jingoEnc3 = jingo.NewStructEncoder(testStruct3{})
	buf := jingo.NewBufferFromPool()
	buf.Reset()
	v := testStruct3{
		Str:"\n",
	}
	jingoEnc3.Marshal(&v, buf)
	fmt.Printf("%q\n",buf)
	buf.Reset()
	buf2, _ := json.Marshal(&v)
	fmt.Printf("%q\n",buf2)
	buf.ReturnToPool()
}
```
out:
```
"{\"str\":\"\n\"}"
"{\"str\":\"\\n\"}"
```

**Comment by kungfusheep** [comment]
We've had some discussion over our treatment of `[]byte`. 

The standard lib behaviour is (from the comments)

```
// Array and slice values encode as JSON arrays, except that
// []byte encodes as a base64-encoded string, 
```

so from our perspective if that behaviour is required that can be achieved with the `stringer` option. 

In the same respect, **escaping** can also be achieved with the `stringer` option.

For raw fields - like in your first example - we can see potential scenarios where that would be useful to directly encode fields. In this case we will be adding a `,raw` tag option, which will direct-write byteslice-like objects (which includes strings) to the buffer, unquoted.

**Comment by kungfusheep** [comment]
fixed by b99ff25d92c16566f8a0a1cf69e4d0cd9bc6033d
